ElectronicObserver/Window/Tools/SenkaViewer/SenkaViewerViewModel.cs

```diff
@@ -54,92 +57,143 @@ public partial class SenkaViewerViewModel : WindowViewModelBase
 
 	public string Today => $"{DropRecordViewerResources.Today}: {DateTime.Now:yyyy/MM/dd}";
 
-	public DataGridViewModel<SenkaRecord> DataGridViewModel { get; set; } = new();
+	public DataGridViewModel<SenkaRecord> DataGridViewModel { get; } = new();
+
+	private DateTime SearchStartTime { get; set; }
+	public string? StatusBarText { get; private set; }
+
+	private static Func<ElectronicObserverContext, Task<DateTime>> MinDateQuery { get; } = EF
+		.CompileAsyncQuery((ElectronicObserverContext db) => db.ApiFiles.Min(f => f.TimeStamp));
 
 	public SenkaViewerViewModel()
 	{
 		SenkaViewer = Ioc.Default.GetRequiredService<SenkaViewerTranslationViewModel>();
 		ResourceRecord = RecordManager.Instance.Resource;
 
-		MinDate = Db.ApiFiles.Min(f => f.TimeStamp);
+		MinDate = new(DateTime.Now.Year, DateTime.Now.Month, 1);
 		MaxDate = DateTime.Now.AddDays(1);
 
 		DateBegin = MinDate.Date;
 		DateEnd = MaxDate.Date;
+
+		Task.Run(async () =>
+		{
+			ElectronicObserverContext db = new();
+
+			DateTime minDate = await MinDateQuery(db);
+
+			if (MinDate < minDate)
```

Comment:
Senka record doesn't load for previous months
Shouldn't the logic be 
```cs
 if (minDate < MinDate)
```

Reply:
Actually totally unrelated.
Previous month data is broken because of a bug in Kancolle API that lasted a day after last maint